pytorch_lightning/overrides/base.py

```diff
@@ -0,0 +1,63 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any
+
+import torch
+
+from pytorch_lightning import LightningModule
+from pytorch_lightning.trainer.states import RunningStage
+from pytorch_lightning.utilities.warnings import WarningCache
+
+warning_cache = WarningCache()
+
+
+class _LightningModuleWrapperBase(torch.nn.Module):
```

Comment:
interesting I think I have never seen "protected" class :]

Reply:
An example that acomes to mind is _LRScheduler from PyTorch :)